packages/automerge-repo/src/synchronizer/CollectionSynchronizer.ts

```diff
@@ -62,7 +62,7 @@ export class CollectionSynchronizer extends Synchronizer {
     const documentId = channelId as unknown as DocumentId
     const docSynchronizer = await this.#fetchDocSynchronizer(documentId)
 
-    docSynchronizer.receiveSyncMessage(peerId, channelId, message)
+    await docSynchronizer.receiveSyncMessage(peerId, channelId, message)
 
     // Initiate sync with any new peers
     const peers = await this.#documentGenerousPeers(documentId)
```

Comment:
Before, this ran in parallel with `receiveSyncMessage` above, which was probably not intended.